### PR TITLE
[TASK_2]About and License viewContrillers and buttons were added.

### PR DESCRIPTION
**Pull request by Noegon**
[TASK_2]About and License viewContrillers and buttons were added. Also old mistakes with autorelease pools and wrong local variables names were corrected.

### CODE REVIEW

File: Calculator/AdditionalControllers/LicenseViewController.h [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+//
+//  Created by user on 16.05.17.
+//  Copyright © 2017 study. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+
+@interface LicenseViewController : UIViewController
+
+- (IBAction)closeLicenseViewController:(UIButton *)sender;
```
Comment (imironenko) [inline Calculator/AdditionalControllers/LicenseViewController.h]: Стоит использовать более корректное наименование, например `closeButtonPressed`, `backButtonPressed`, `handlerCloseAction`

File: Calculator/ViewController.m [hunk 2]
```diff
@@ -7,50 +7,111 @@
...
 - (void)viewDidLoad {
     [super viewDidLoad];
     
-    // we should to create an instance of UIGestureRecognizer to use it
-    // (if we didn't do it in interface builder)
-    // And, ofcourse, we should to release it in dealloc method if we don't use ARC
-//    UIGestureRecognizer *swipeRec = [[UIGestureRecognizer alloc] initWithTarget:self action:@selector(handleSwipeRecognizer:)];
-//    [[self nameTextField] addGestureRecognizer:swipeRec];
+    self.navigationController.navigationBar.backgroundColor = [UIColor grayColor];
+    UIBarButtonItem *aboutBarButton = [[UIBarButtonItem alloc] initWithTitle:@"About" style:UIBarButtonItemStylePlain target:self action:@selector(aboutButtonTouched:)];
```
Comment (imironenko) [inline Calculator/ViewController.m]: утечка памяти

File: Calculator/ViewController.m [hunk 3]
```diff
@@ -7,50 +7,111 @@
...
+- (void)switchCalculationButtonsEnabled:(BOOL)areBUttonsEnabled {
+    if (!areBUttonsEnabled) {
+        for (UIButton *view in self.view.subviews) {
+            view.enabled = NO;
+        }
+        self.clearButton.enabled = YES;
+        self.aboutButton.enabled = YES;
     } else {
-        self.digitInsertionField.text = result;
+        for (UIButton *view in self.view.subviews) {
```
Comment (imironenko) [inline Calculator/ViewController.m]: неккоректное обработка, на view могут присутствовать не только buttons, используй свойство `digitButtonsArray`

File: Calculator/ViewController.m [hunk 4]
```diff
@@ -7,50 +7,111 @@
...
-    // we should to create an instance of UIGestureRecognizer to use it
-    // (if we didn't do it in interface builder)
-    // And, ofcourse, we should to release it in dealloc method if we don't use ARC
-//    UIGestureRecognizer *swipeRec = [[UIGestureRecognizer alloc] initWithTarget:self action:@selector(handleSwipeRecognizer:)];
-//    [[self nameTextField] addGestureRecognizer:swipeRec];
+    self.navigationController.navigationBar.backgroundColor = [UIColor grayColor];
+    UIBarButtonItem *aboutBarButton = [[UIBarButtonItem alloc] initWithTitle:@"About" style:UIBarButtonItemStylePlain target:self action:@selector(aboutButtonTouched:)];
+    self.navigationItem.leftBarButtonItem = aboutBarButton;
+    
+    UIBarButtonItem *licenseBarButton = [[UIBarButtonItem alloc] initWithTitle:@"License" style:UIBarButtonItemStylePlain target:self action:@selector(licenseButtonTouched:)];
```
Comment (imironenko) [inline Calculator/ViewController.m]: утечка памяти

**Comment by Noegon** [comment]
Все замечания исправил, коммитнул.18.05.2017, 00:33, "Igor Mironenko" <notifications@github.com>:@imironenko commented on this pull request.



In Calculator/ViewController.m:
>  @end
 
 @implementation ViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
     
-    // we should to create an instance of UIGestureRecognizer to use it
-    // (if we didn't do it in interface builder)
-    // And, ofcourse, we should to release it in dealloc method if we don't use ARC
-//    UIGestureRecognizer *swipeRec = [[UIGestureRecognizer alloc] initWithTarget:self action:@selector(handleSwipeRecognizer:)];
-//    [[self nameTextField] addGestureRecognizer:swipeRec];
+    self.navigationController.navigationBar.backgroundColor = [UIColor grayColor];
+    UIBarButtonItem *aboutBarButton = [[UIBarButtonItem alloc] initWithTitle:@"About" style:UIBarButtonItemStylePlain target:self action:@selector(aboutButtonTouched:)];
+    self.navigationItem.leftBarButtonItem = aboutBarButton;
+    
+    UIBarButtonItem *licenseBarButton = [[UIBarButtonItem alloc] initWithTitle:@"License" style:UIBarButtonItemStylePlain target:self action:@selector(licenseButtonTouched:)];

утечка памяти

File: Calculator/ViewController.m [hunk 5]
```diff
@@ -7,50 +7,133 @@
...
 - (void)viewDidLoad {
     [super viewDidLoad];
     
-    // we should to create an instance of UIGestureRecognizer to use it
-    // (if we didn't do it in interface builder)
-    // And, ofcourse, we should to release it in dealloc method if we don't use ARC
-//    UIGestureRecognizer *swipeRec = [[UIGestureRecognizer alloc] initWithTarget:self action:@selector(handleSwipeRecognizer:)];
-//    [[self nameTextField] addGestureRecognizer:swipeRec];
+    self.navigationController.navigationBar.backgroundColor = [UIColor grayColor];
+    _aboutBarButton = [[UIBarButtonItem alloc] initWithTitle:@"About" style:UIBarButtonItemStylePlain target:self action:@selector(aboutButtonTouched:)];
```
Comment (imironenko) [inline Calculator/ViewController.m:46]: используй обращение через свойство `self.aboutBarButton`

File: Calculator/ViewController.m [hunk 6]
```diff
@@ -7,50 +7,133 @@
...
+    self.navigationItem.leftBarButtonItem = _aboutBarButton;
+    [_aboutBarButton release];
```
Comment (imironenko) [inline Calculator/ViewController.m:48]: при текущей реализации - leak

File: Calculator/ViewController.m [hunk 7]
```diff
@@ -7,50 +7,133 @@
...
+    
+    
+    _licenseBarButton = [[UIBarButtonItem alloc] initWithTitle:@"License" style:UIBarButtonItemStylePlain target:self action:@selector(licenseButtonTouched:)];
```
Comment (imironenko) [inline Calculator/ViewController.m:51]: внеси правки, в соответствии с комментарием выше

File: Calculator/ViewController.m [hunk 8]
```diff
@@ -7,50 +7,133 @@
...
+    self.navigationItem.rightBarButtonItem = _licenseBarButton;
+    [_licenseBarButton release];
```
Comment (imironenko) [inline Calculator/ViewController.m:53]: при текущей реализации - leak

**Pull request merged**